withReduxSaga/withReduxSaga.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import withRedux from 'next-redux-wrapper';
+
+export default params => (InnerComponent, actions) => {
+  class ReduxContainer extends Component {
+    static async getInitialProps({ store, isServer, ...rest }) {
+      if (isServer) {
+        const action = actions.server || actions;
+        const rootTask = store.runSagas();
+        store.dispatch(Object.assign({}, action, { isServer }, { query: rest.query }));
+        store.close();
+        await rootTask.done.then(() => store.reset());
+      } else {
+        const action = actions.client || actions;
+        store.runSagas();
+        store.dispatch(Object.assign({}, action, { isServer }, { query: rest.query }));
+      }
+    }
+
+    constructor(props) {
+      super(props);
+      params.store.runSagas();
+    }
+
+    render() {
+      return <InnerComponent {...this.props} />;
+    }
+  }
+
+  return withRedux(params.makeStore)(ReduxContainer);
```

Comment:
@batusai513 : the params.makeStore is correct. `withRedux` takes either a function or an configuration object. If its an object, then there has to be a field `createStore` and there are a few other options - the one I need is `storeKey`. This sets the redux store key to something that I can control and so can lookup in `window` object to see if the store exists already.